### PR TITLE
Allow amber to correctly handle decoupled molecule

### PR DESCRIPTION
**Pull request by xiki-tempula**
This PR allows the config factory to correctly handle the decoupled molecule and generate the correct config.

### CODE REVIEW

File: python/BioSimSpace/Sandpit/Exscientia/Align/_squash.py [hunk 1]
```diff
@@ -442,7 +454,13 @@ def _squashed_atom_mapping_molecule(
         The number of squashed atoms that correspond to the squashed molecule.
     """
     if molecule.isDecoupled():
-        if dummies and not is_lambda1:
+        # Check if the state 0 is coupled
+        lambda0_coupled = _check_decouple(molecule)
```
Comment (msuruzhon) [inline python/BioSimSpace/Sandpit/Exscientia/Align/_squash.py:458]:
```python
decoupled_lambda = not _check_decouple(molecule)
if decoupled_lambda is is_lambda1:
  return {
          offset_merged + i: offset_squashed + i for i in range(molecule.nAtoms())
      }, molecule.nAtoms()
else:
  return {}, molecule.nAtoms()
```
?

File: python/BioSimSpace/Sandpit/Exscientia/Protocol/_config.py [hunk 2]
```diff
@@ -223,12 +224,12 @@ def generateAmberConfig(self, extra_options=None, extra_lines=None):
             protocol_dict["ntf"] = 2  # Don't calculate forces for constrained bonds.
 
         # PBC.
-        if not self._has_box or not self._has_water:
-            protocol_dict["ntb"] = 0  # No periodic box.
-            protocol_dict["cut"] = "999."  # Non-bonded cut-off.
-        else:
+        if self._has_box:
```
Comment (msuruzhon) [inline python/BioSimSpace/Sandpit/Exscientia/Protocol/_config.py:227]: Wondering whether we should propagate that to the GROMACS config as well...

File: python/BioSimSpace/Sandpit/Exscientia/Protocol/_config.py [hunk 3]
```diff
@@ -336,12 +339,15 @@ def generateAmberConfig(self, extra_options=None, extra_lines=None):
             protocol_dict["icfe"] = 1  # Free energy mode.
             protocol_dict["ifsc"] = 1  # Use softcore potentials.
             protocol_dict["ntf"] = 1  # Remove SHAKE constraints.
-            protocol = [str(x) for x in self.protocol.getLambdaValues()]
+            LambdaValues = self.protocol.getLambdaValues(type="dataframe")
```
Comment (msuruzhon) [inline python/BioSimSpace/Sandpit/Exscientia/Protocol/_config.py:342]: CamelCase variable names are a bit confusing

File: test/Sandpit/Exscientia/Protocol/test_config.py [hunk 4]
```diff
@@ -269,3 +283,27 @@ def test_sc_parameters(self, system):
         with open(f"{freenrg._work_dir}/lambda_6/gromacs.mdp", "r") as f:
             mdp_text = f.read()
             assert "sc-alpha = 0.5" in mdp_text
+
+
+class TestAMBERABFE:
```
Comment (msuruzhon) [inline test/Sandpit/Exscientia/Protocol/test_config.py:288]: Isn't that ASFE?

File: python/BioSimSpace/Sandpit/Exscientia/Protocol/_config.py (hunk 2 above)
Comment (xiki-tempula) [reply python/BioSimSpace/Sandpit/Exscientia/Protocol/_config.py:227]: Maybe we just leave it like this for now. I think Lester might have some strong feeling about Gromacs where he need to handle water differently.

**Pull request merged**